src/components/characters/CharacterToggle.jsx

```diff
@@ -0,0 +1,24 @@
+/* eslint-disable max-len */
+import React from 'react';
+import { useAvailableAPIs, useSetSelectedApi } from '../../state/CharacterProvider';
+
+
+function CharacterToggle() {
+  const availableApis = useAvailableAPIs();
+  const setSelectedApi = useSetSelectedApi();
+
+  const handleChange = ({ target }) => setSelectedApi(target.value);
+  
+  return (
+    <ul>
+      {availableApis .map(api => (
+        <li key={api}>
+          <label>{api}</label>
+          <input type="radio" name="api" value={api} onChange={handleChange} />
```

Comment:
checked={selectedApi=== api}
you could've passed this selectedApi from your context and caused a boolean that would add to the user experience. (the button would be marked on load). Not a big detail but one that people might expect.